### DISCUSSION TITLE
Pyvista, vtk and GIL

### DISCUSSION THREAD

**Discussion by igorbunova** [pyvista/pyvista#4267] [Q&A]
I'm using pyvista 0.36.1 with vtk 9.2.6 in the python web application.
I wonder if there is any way not to have all threads locked when vtk native calls happen.
Some of my operations are pretty long:
Clipping with Function: 100%  [00:46<00:00]
Resampling array Data from a Passed Mesh onto Mesh: 100% [01:07<00:00]
And during this time the entire application is blocked.

Researching for this problem I found few discussions regarding this issue and various options to try on vtk build from sources:
VTK_PYTHON_FULL_THREADSAFE
VTK_NO_PYTHON_THREADS
VTK_USE_PARALLEL
VTK_USE_MPI
I tried various combinations, and I'm not sure these options (except VTK_USE_MPI, because it is documented) are relevant.
And finally the problem remains the same.
Does anybody know how to resolve this issue?

**Comment by banesullivan** [comment]
PyVista makes mesh objects serializable (pickle-able) across threads, so you could put the expensive operation in a background thread and have it update the visualization or web app when complete to prevent blocking the main app thread.
